code/game/jobs/job_exp.dm

```diff
@@ -0,0 +1,234 @@
+		if(exp_sec > 0)
+			return_text += "<BR>- Security: " + get_exp_format(exp_sec) + " (" + num2text(round(exp_sec/exp_gen*100)) + "%)"
+		if(exp_sci > 0)
+			return_text += "<BR>- Science: " + get_exp_format(exp_sci) + " (" + num2text(round(exp_sci/exp_gen*100)) + "%)"
+		if(exp_eng > 0)
+			return_text += "<BR>- Engineering: " + get_exp_format(exp_eng) + " (" + num2text(round(exp_eng/exp_gen*100)) + "%)"
+		if(exp_med > 0)
+			return_text += "<BR>- Medical: " + get_exp_format(exp_med) + " (" + num2text(round(exp_med/exp_gen*100)) + "%)"
+		if(exp_sup > 0)
+			return_text += "<BR>- Support: " + get_exp_format(exp_sup) + " (" + num2text(round(exp_sup/exp_gen*100)) + "%)"
+		if(exp_sil > 0)
+			return_text += "<BR>- Silicon: " + get_exp_format(exp_sil) + " (" + num2text(round(exp_sil/exp_gen*100)) + "%)"
+		if(exp_ant > 0)
+			return_text += "<BR>- Special: " + get_exp_format(exp_ant) + " (" + num2text(round(exp_ant/exp_gen*100)) + "%)"
+		if(exp_gho > 0)
+			return_text += "<BR>- Ghost: " + get_exp_format(exp_gho) + " (" + num2text(round(exp_gho/exp_gen*100)) + "%)"
+		if(exp_imm > 0)
+			return_text += "<BR>- Grandfathered: YES"
+		var/list/jobs_locked = list()
+		var/list/jobs_unlocked = list()
+		for(var/datum/job/job in job_master.occupations)
+			if(job.exp_requirements && job.exp_type)
+				if(!job_is_xp_locked(job.title))
+					continue
+				else if(has_exp_for_job(mob, job.title))
+					jobs_unlocked += "<BR>- " + job.title
+				else
+					jobs_locked += "<BR>- " + job.title + " (" + get_exp_format(text2num(play_records[job.exp_type])) + " / " + get_exp_format(job.exp_requirements) + " [job.exp_type] EXP)"
+		if(jobs_unlocked.len)
+			return_text += "<BR><BR>Jobs Unlocked: "
+			for(var/text in jobs_unlocked)
+				return_text += text
+		if(jobs_locked.len)
+			return_text += "<BR><BR>Jobs Not Unlocked: "
+			for(var/text in jobs_locked)
+				return_text += text
+		return return_text
+	else
+		return "[src] has no EXP records."
+
+/client/proc/get_exp_general()
+	var/list/play_records = params2list(prefs.exp)
+	var/exp_gen = text2num(play_records["gen"])
+	return get_exp_format(exp_gen)
+
+/client/proc/get_exp_format(var/expnum)
+	if(expnum > 0)
+		return num2text(round(expnum / 60)) + "h"
+	else
+		return "0h"
+
+/proc/update_exp(var/minutes, var/announce_changes = 0)
+	if(!establish_db_connection())
+		return -1
+	for(var/client/C in clients)
+		if(C.inactivity < (10 MINUTES))
+
+			var/DBQuery/exp_read = dbcon.NewQuery("SELECT exp FROM [format_table_name("player")] WHERE ckey='[C.ckey]'")
+			exp_read.Execute()
+			var/list/play_records = list()
+			while(exp_read.NextRow())
+				play_records = params2list(exp_read.item[1])
+
+			for(var/rtype in list("gen","com","sec","sci","eng","med","sup","sil","ant", "gho", "imm"))
+				if(!text2num(play_records[rtype]))
+					play_records[rtype] = 0
+				else
+					play_records[rtype] = text2num(play_records[rtype])
+
+			if(C.mob.stat == CONSCIOUS && C.mob.mind.assigned_role)
```

Comment:
:-1:

if you die you shouldn't lose the experience, fuck off with this shit its about learnning to play not whether you lost those 2 hours of experience in a freak accident